File: healthchecks/payment-service/src/main/java/com/varunu28/paymentservice/repository/PaymentRepository.java
package com.varunu28.paymentservice.repository;

import com.varunu28.paymentservice.model.Payment;
import java.util.UUID;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface PaymentRepository extends CrudRepository<Payment, UUID> {
}

File: saga/payment-service/src/main/kotlin/com/saga/paymentservice/PaymentRepository.kt
package com.saga.paymentservice

import jakarta.persistence.LockModeType
import org.springframework.data.jpa.repository.Lock
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.math.BigInteger

@Repository
interface PaymentRepository : CrudRepository<Payment, BigInteger> {

    @Lock(LockModeType.PESSIMISTIC_WRITE)
    fun findByOrderId(orderId: BigInteger): Payment?
}